1.28.0

matteobruni opened the pull request:
# New Features
- Added a Web Component npm package to fix this issue VincentGarreau/particles.js#274 reported by @erispre and @abhimanyud3dx
- Added a Solid.js Component npm package

# Bug Fixes
- Fixed issue #1594 
- Fixed issue #1587

matteobruni committed:
build: angular updated

matteobruni committed:
build: restored node.js ci

matteobruni committed:
build: trying esbuild

matteobruni committed:
Merge branch 'main' into v1

matteobruni committed:
build: trying esbuild

matteobruni committed:
build: removed esbuild, bigger bundle size

matteobruni committed:
feat: added confetti cannon preset
chore(presets): removed some unused presets
build: updated dependencies

matteobruni committed:
chore(demo): renamed demo/main to demo/vanilla

matteobruni committed:
build: added images to presets demo

matteobruni committed:
build: renamed basic preset to links

matteobruni committed:
build: migrating presets to new architecture

matteobruni committed:
build: migrated other presets

matteobruni committed:
build: more presets migrated

matteobruni committed:
build: all presets migrated

matteobruni committed:
build: prettier

matteobruni committed:
build: build files

matteobruni committed:
build: fixed links preset and demos

matteobruni committed:
Update nodejs.yml

matteobruni committed:
feat: added distance to move.attract options
build: some improvements to exports
build: improved some presets

matteobruni committed:
build: fixed errors from previous commit

matteobruni committed:
build: prettier

matteobruni committed:
build: fixed previous commits errors

matteobruni committed:
build: removed unused imports

matteobruni committed:
feat: preparing simpler options for confetti preset

matteobruni committed:
feat: prepared confetti preset

matteobruni committed:
docs: improving documentation

matteobruni committed:
docs: improving documentation

matteobruni committed:
docs: improving documentation

matteobruni committed:
build: deps updated

matteobruni committed:
feat: adding SolidJS component

matteobruni committed:
chore(solid): improving solid component, now project builds

matteobruni committed:
chore(solid): improving solid component, now project builds

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
chore(solid): improving solid component

matteobruni committed:
feat: added web component

matteobruni committed:
chore(component): improved web component

matteobruni committed:
build: preparing web components demo

matteobruni committed:
build: improving web component

matteobruni committed:
build: fixed web component

matteobruni committed:
Merge pull request #1570 from matteobruni/web-component

feat: added web component

matteobruni committed:
build: improving web component

matteobruni committed:
build: improving canvas resize

matteobruni committed:
build: preparing web-component for publishing

matteobruni committed:
build: preparing web-component for publishing

matteobruni committed:
Merge branch 'v1' into solid-js

matteobruni committed:
chore(components): fixed solid component

matteobruni committed:
Merge pull request #1556 from matteobruni/solid-js

feat: adding SolidJS component

matteobruni committed:
build: updated angular component deps

matteobruni committed:
build: updated inferno and jquery component deps

matteobruni committed:
build: updated preact component deps

matteobruni committed:
build: updated solid component deps

matteobruni committed:
build: updated react component deps

matteobruni committed:
build: updated riot component deps

matteobruni committed:
build: updated svelte component deps

matteobruni committed:
build: updated vue component deps

matteobruni committed:
build: updated web component deps

matteobruni committed:
build: updated core packages deps, broken typedoc with ts4.3

matteobruni committed:
build: updated presets and shapes deps

matteobruni committed:
build: updated templates

matteobruni committed:
build: updated angular demo deps

matteobruni committed:
build: updaed editor demo deps

matteobruni committed:
build: updaed inferno demo deps

matteobruni committed:
build: updaed jquery demo deps

matteobruni committed:
build: updaed preact demo deps

matteobruni committed:
build: updaed react demo deps

matteobruni committed:
build: updaed riot demo deps

matteobruni committed:
build: updaed solid demo deps

matteobruni committed:
build: updaed svelte demo deps

matteobruni committed:
build: updaed vanilla and vue demo deps

matteobruni committed:
build: updated web components demo deps

matteobruni committed:
Merge branch 'main' into v1

matteobruni committed:
build: updates after merge

ghost commented:
## DeepCode's analysis on [#91e485](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/97e944d6da53da06d78f4c28641e8e6fdccc33c3/matteobruni/tsparticles/91e485d0de41e14ad7039d87ccee038bc2e254cb/pr/_/%2F/code/?utm_source=gh_review&c=0&w=3&i=1&) found:

- :warning: **3** warnings, :information_source: **1** minor issue. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">No catch method for promise. This may result in an unhandled promise rejection. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/91e485d0de41e14ad7039d87ccee038bc2e254cb/components/web/build.ts#L45">build.ts:45</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/97e944d6da53da06d78f4c28641e8e6fdccc33c3/matteobruni/tsparticles/91e485d0de41e14ad7039d87ccee038bc2e254cb/pr/_/%2Fcomponents%2Fweb%2Fbuild.ts/javascript%2Fdc_interfile_project%2FPromiseNotCaughtNode/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Consider using csurf middleware for your Express app to protect against CSRF attacks. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/91e485d0de41e14ad7039d87ccee038bc2e254cb/demo/web/app.js#L5">app.js:5</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/97e944d6da53da06d78f4c28641e8e6fdccc33c3/matteobruni/tsparticles/91e485d0de41e14ad7039d87ccee038bc2e254cb/pr/_/%2Fdemo%2Fweb%2Fapp.js/javascript%2Fdc_interfile_project%2FUseCsurfForExpress/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">This endpoint handler performs a file system operation and does not use a rate-limiting mechanism. It may enable the attackers to perform Denial-of-service attacks. Consider using a rate-limiting middleware such as express-limit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/91e485d0de41e14ad7039d87ccee038bc2e254cb/demo/web/app.js#L20">app.js:20</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/91e485d0de41e14ad7039d87ccee038bc2e254cb/demo/vanilla/app.js#L41">app.js:41</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/91e485d0de41e14ad7039d87ccee038bc2e254cb/demo/vanilla/app.js#L45">app.js:45</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/97e944d6da53da06d78f4c28641e8e6fdccc33c3/matteobruni/tsparticles/91e485d0de41e14ad7039d87ccee038bc2e254cb/pr/_/%2Fdemo%2Fweb%2Fapp.js/javascript%2Fdc_interfile_project%2FNoRateLimitingForExpensiveWebOperation/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/97e944d6da53da06d78f4c28641e8e6fdccc33c3/matteobruni/tsparticles/91e485d0de41e14ad7039d87ccee038bc2e254cb/pr/_/%2F/code/?utm_source=gh_review&c=0&w=3&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

matteobruni committed:
fix: fixed issue #1594

matteobruni committed:
Merge pull request #1596 from matteobruni/issue-1594

Fix Issue 1594

matteobruni committed:
fix: debounce window resize, check if #1587 is fixed
feat: added firefly preset

matteobruni committed:
fix: fixed #1587

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release

matteobruni committed:
build: preparing docs for release
chore(solid): the local component was used, the external one is still broken

matteobruni committed:
build: solid component is now private until a fix is found, so the 1.28.0 can be released

matteobruni committed:
build: web components demo has less files

matteobruni committed:
chore(firefly): improving firefly preset

matteobruni committed:
build: improving docs for release

matteobruni committed:
build: try fix solid

matteobruni committed:
build: fixed solid component

matteobruni committed:
build: updated some preset demo images

matteobruni committed:
fix: fixed confetti preset
build: fixed confetti preset docs

matteobruni committed:
build: added fire preset docs

matteobruni committed:
build: added all presets README.md

matteobruni committed:
fix: changed click handler, now moved to Container class but kept compatibility

matteobruni committed:
build: updated deps before release

matteobruni committed:
chore(release): published new version

 - ng-particles@2.11.0
 - inferno-particles@1.11.0
 - jquery-particles@1.28.0
 - preact-particles@1.28.0
 - react-tsparticles@1.28.0
 - riot-particles@1.2.0
 - solid-particles@1.11.0
 - svelte-particles@1.11.0
 - particles.vue@2.11.0
 - particles.vue3@1.11.0
 - web-particles@1.1.0
 - tsparticles-editor@1.28.0
 - tsparticles@1.28.0
 - ng-particles-demo@1.13.0
 - editor-demo@1.13.0
 - inferno-particles-demo@1.11.0
 - jquery-particles-demo@1.13.0
 - preact-particles-demo@1.13.0
 - react-tsparticles-demo@1.13.0
 - riot-particles-demo@1.2.0
 - solid-particles-demo@1.11.0
 - svelte-demo@1.11.0
 - tsparticles-demo@1.13.0
 - particles.vue-demo@1.13.0
 - particles.vue3-demo@1.11.0
 - web-particles-demo@1.13.0
 - tsparticles-preset-big-circles@1.13.0
 - tsparticles-preset-bubbles@1.13.0
 - tsparticles-preset-confetti@1.13.0
 - tsparticles-preset-fire@1.13.0
 - tsparticles-preset-firefly@1.13.0
 - tsparticles-preset-fountain@1.13.0
 - tsparticles-preset-links@1.13.0
 - tsparticles-preset-sea-anemone@1.13.0
 - tsparticles-preset-snow@1.13.0
 - tsparticles-preset-stars@1.13.0
 - tsparticles-shape-bubble@1.13.0
 - tsparticles-shape-confetti@1.13.0
 - tsparticles-shape-heart@1.13.0
 - tsparticles-shape-multiline-text@1.13.0
 - tsparticles-shape-rounded-rect@1.13.0
 - tsparticles-shape-spiral@1.13.0
 - cra-template-particles-typescript@1.11.0
 - cra-template-particles@1.11.0
 - tsparticles-website@1.28.0

matteobruni committed:
build: removed tests from riot component

matteobruni committed:
build: updated website
